ui/app/helpers/utils/fetch-with-cache.test.js

```diff
@@ -135,4 +131,43 @@ describe('Fetch with cache', function () {
       { message: 'fetchWithCache only supports JSON responses' },
     )
   })
+
+  it('should correctly cache responses from interwoven requests', async function () {
```

Comment:
I did test this on `develop` to ensure that it fails. It doesn't work exactly as written because the cache object is different, but after adjusting this, it did fail for me consistently (the `/foo` response was always missing).